### PR TITLE
linux-firmware: move MediaTek firmware

### PR DESCRIPTION
**Pull request by Sapein**
Currently only some MediaTek firmware is moved.

#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by leahneukirchen** [comment]
Should also move
```
linux-firmware-20220209_1	/usr/lib/firmware/vpu_d.bin -> /usr/lib/firmware/mediatek/mt8173/vpu_d.bin
linux-firmware-20220209_1	/usr/lib/firmware/vpu_p.bin -> /usr/lib/firmware/mediatek/mt8173/vpu_p.bin
```

**Comment by leahneukirchen** [comment]
... that doesnt make sense. you need to move the symlinks, the targets are already there.

File: srcpkgs/linux-firmware/template [hunk 1]
```diff
@@ -94,6 +94,10 @@ linux-firmware-network_package() {
 		vmove usr/lib/firmware/libertas
 		vmove usr/lib/firmware/mrvl
 		vmove "usr/lib/firmware/mt*.bin"
+		vmove usr/lib/firmware/mediatek
+		vmove usr/lib/firmware/mediatek/mt8173/vpu_d.bin
+		vmove usr/lib/firmware/mediatek/mt8173/vpu_p.bin
+		vmove usr/lib/firmware/mediatek
```
Comment (sgn) [inline srcpkgs/linux-firmware/template:100]: This looks sketchy

**Comment by Sapein** [comment]
> ... that doesnt make sense. you need to move the symlinks, the targets are already there.

Apologies, I had misread what you meant, but I'm still somewhat confused as to what is needed to be done. I'm somewhat new with this.

**Comment by nicegamer7** [comment]
Is there anything blocking this? It'd be nice to get this merged.

**Comment by JamiKettunen** [comment]
I believe @leahneukirchen's comment from March applies, `/usr/lib/firmware/vpu_{d,p}.bin` are still not being moved to the `-network` subpackage as well.

**Comment by q66** [comment]
fwiw i don't think a bunch of the mediatek firmware should go to -network (notably the vpu_* stuff as that's not network-related at all)

there is also a lot of firmware in the main package that *should* go to -network though

**Comment by nicegamer7** [comment]
From a cursory glance, it looks like all the files in in the mediatek folder are network related, but the directories aren't.

**Comment by JamiKettunen** [comment]
[Here's how Chimera Linux does it currently](https://github.com/chimera-linux/cports/blob/6d47cc1/main/firmware-linux/template.py#L82-L85), maybe we could do something similar? would allow us to separate the network-related and ARM SoC specific files by moving the ladder to a `linux-firmware-mediatek` subpackage and then bringing rest of `mediatek` dir and `mt7*.bin` into `linux-firmware-network`.

Also this commit currently is very vague: `linux-firmware: move MediaTek firmware`, move them *where*? obviously into `-network` subpackage but it could be more specific for commit history readability

@Sapein are you interested in continuing (to fix) this PR?

**Comment by Sapein** [comment]
> [Here's how Chimera Linux does it currently](https://github.com/chimera-linux/cports/blob/6d47cc1/main/firmware-linux/template.py#L82-L85), maybe we could do something similar? would allow us to separate the network-related and ARM SoC specific files by moving the ladder to a `linux-firmware-mediatek` subpackage and then bringing rest of `mediatek` dir and `mt7*.bin` into `linux-firmware-network`.
> 
> Also this commit currently is very vague: `linux-firmware: move MediaTek firmware`, move them _where_? obviously into `-network` subpackage but it could be more specific for commit history readability
> 
> @Sapein are you interested in continuing (to fix) this PR?

Hey, sorry for the disappearance, but yes I am still interested in continuing this PR and attempting to fix it. Part of the issue is that I am a bit unfamiliar with Packaging in general (and Void Linux packaging in particular), so I'm unsure how to go about fixing it. I can adjust the commit message to be something like: `move MediaTek firmware into linux-firmware-network` if desired, although I'm not sure how to best adjust the commit message.

**Pull request closed without merging**